lnwallet/rpcwallet/remote_signer.go

```diff
@@ -91,3 +98,162 @@ type RemoteSignerRequests interface {
+	timeout time.Duration
+}
+
+// NewInboundRemoteSigner creates a new InboundRemoteSigner instance.
+// The function sets up a connection to the remote signer node.
+// The returned function is a cleanup function that should be called to close
+// the connection when the remote signer is no longer needed.
+func NewInboundRemoteSigner(rpcHost string, tlsCertPath string,
+	macaroonPath string,
+	timeout time.Duration) (*InboundRemoteSigner, func(), error) {
+
+	rpcConn, err := connect(rpcHost, tlsCertPath, macaroonPath, timeout)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error connecting to the remote "+
+			"signing node through RPC: %v", err)
+	}
+
+	cleanUp := func() {
+		rpcConn.Close()
+	}
+
+	remoteSigner := &InboundRemoteSigner{
+		SignerClient:    signrpc.NewSignerClient(rpcConn),
+		WalletKitClient: walletrpc.NewWalletKitClient(rpcConn),
+		rpcHost:         rpcHost,
+		tlsCertPath:     tlsCertPath,
+		macaroonPath:    macaroonPath,
+		timeout:         timeout,
+	}
+
+	return remoteSigner, cleanUp, nil
+}
+
+// Run feeds lnd with the incoming stream set up by an outbound remote signer
+// and then blocks until the stream is closed. Lnd can then send any requests to
+// the remote signer through the stream.
+//
+// NOTE: This is part of the RemoteSigner interface.
+func (*InboundRemoteSigner) Run(_ StreamServer) error {
+	// If lnd has been configured to use an inbound remote signer, it should
+	// not allow an outbound remote signer to connect.
+	return errors.New("not supported when remotesigner.signerrole is set " +
+		"to \"watchonly-inbound\"")
+}
+
+// Ready blocks and returns nil when the remote signer is ready to accept
+// requests.
+//
+// NOTE: This is part of the RemoteSigner interface.
+func (r *InboundRemoteSigner) Ready() error {
+	// The inbound remote signer is ready as soon we have connected to the
+	// remote signer node in the constructor. Therefore, we always return
+	// nil here to signal that we are ready.
+	return nil
+}
+
+// Ping verifies that the remote signer is still responsive.
+//
+// NOTE: This is part of the RemoteSigner interface.
+func (r *InboundRemoteSigner) Ping(timeout time.Duration) error {
+	conn, err := connect(
+		r.rpcHost, r.tlsCertPath, r.macaroonPath, timeout,
+	)
+	if err != nil {
+		return fmt.Errorf("error connecting to the remote "+
+			"signing node through RPC: %v", err)
+	}
+
+	return conn.Close()
+}
+
+// Timeout returns the set connection timeout for the remote signer.
+//
+// NOTE: This is part of the RemoteSigner interface.
+func (r *InboundRemoteSigner) Timeout() time.Duration {
+	return r.timeout
+}
+
+// A compile time assertion to ensure InboundRemoteSigner meets the
+// RemoteSigner interface.
+var _ RemoteSigner = (*InboundRemoteSigner)(nil)
+
+// connect tries to establish an RPC connection to the given host:port with the
+// supplied certificate and macaroon.
+func connect(hostPort, tlsCertPath, macaroonPath string,
```

Comment:
This was due to that the name that this is later changed to was used in the healthcheck and the `RPCKeyRing`, and didn't want to include changing of those in this commit.

But I've now updated so that this function is a struct member instead, which removes that issue :).